fastapi/models/fastapi_endpoint.py

```diff
@@ -201,8 +201,24 @@ def get_app(self, root_path):
             return None
         app = FastAPI()
         app.mount(record.root_path, record._get_app())
+        self._clear_fastapi_exception_handlers(app)
         return ASGIMiddleware(app)
 
+    def _clear_fastapi_exception_handlers(self, app: FastAPI) -> None:
+        """
+        Clear the exception handlers of the given fastapi app.
+
+        This method is used to ensure that the exception handlers are handled
+        by odoo and not by fastapi. We therefore need to remove all the handlers
+        added by default when instantiating a FastAPI app. Since apps can be
+        mounted recursively, we need to apply this method to all the apps in the
+        mounted tree.
```

Comment:
is this needed when we mount a sub application too?

Reply:
> is this needed when we mount a sub application too?

The method is recursive. It walks across all the sub applications to clear the exception handlers. You shouldn't have to worry about this particular treatment.